up.go

```diff
@@ -453,22 +453,25 @@ func (e *Editor) kill() {
 	e.value = e.value[:e.cursor]
 }
 
-func (e *Editor) killWord() {
-	if e.cursor > 0 {
-		foundBeginingOfWord := false
-		cursor := e.cursor - 1
-		for !foundBeginingOfWord {
-			if cursor == 0 || (e.value[cursor] != ' ' && e.value[cursor-1] == ' ') {
-				foundBeginingOfWord = true
-			} else {
-				cursor--
-			}
+// unixWordRubout removes the part of the word on the left of the cursor. A word is
+// delimited by whitespaces.
+// The term `unix-word-rubout` comes from `readline` (see `man 3 readline`)
+func (e *Editor) unixWordRubout() {
+	if e.cursor <= 0 {
+		return
+	}
+	pos := e.cursor - 1
+	for {
+		if pos == 0 || (e.value[pos] != ' ' && e.value[pos-1] == ' ') {
+			break
 		}
-		afterCursor := e.value[e.cursor:]
-		e.value = e.value[:cursor]
-		e.value = append(e.value, afterCursor...)
-		e.cursor = cursor
+		pos--
 	}
+	afterCursor := e.value[e.cursor:]
+	e.killspace = append(e.killspace[:0], e.value[pos:e.cursor]...)
+	e.value = e.value[:pos]
+	e.value = append(e.value, afterCursor...)
```

Comment:
I think we can reorder it a bit and shorten (see also ["Go slice tricks"](https://github.com/golang/go/wiki/SliceTricks)); this gives the two lines nice symmetry, making it arguably easier to see that what one line taketh, the other line glueth the rest together.

```suggestion
	e.killspace = append(e.killspace[:0], e.value[pos:e.cursor]...)
	e.value = append(e.value[:pos], e.value[e.cursor:]...)
```